tensorboard/data/server/BUILD

```diff
@@ -144,9 +145,32 @@ genrule(
         "//tensorboard/data/proto:proto_srcs",
         "//tensorboard/plugins/audio:proto_srcs",
         "//tensorboard/plugins/image:proto_srcs",
+        "@com_google_protobuf//:well_known_protos",
     ],
     outs = _genproto_files,
-    cmd = "PROTOC=$(execpath @com_google_protobuf//:protoc) $(execpath :gen_protos_tool) $(RULEDIR)",
+    cmd = """
+        die() { printf >&2 'fatal: %s\\n' "$$@"; exit 1; }
+        export PROTOC=$(execpath @com_google_protobuf//:protoc)
+        # The `:well_known_protos` fileset has source files in the `src/`
+        # directory under the genfiles tree for its workspace. Bazel variable
+        # expansion doesn't provide a way to get that directly (it's not a
+        # target), but we can extract it from the files themselves.
+        f=
+        for f in $(locations @com_google_protobuf//:well_known_protos); do
+            break  # just take one arbitrarily
+        done
+        [ -n "$${f}" ] || die 'no well-known protos in fileset'
+        dir="$$(dirname "$${f}")"
+        while true; do
```

Comment:
This while loop feels like a bit of a complicated way to effectively just remove the suffix `/src/.*?` and replace it with `/src`.  Maybe consider just doing string munging for that?